src/firmware/main/math/polynomial_2d.c

```diff
@@ -0,0 +1,178 @@
+#include "firmware/main/math/polynomial_2d.h"
+
+#include <assert.h>
+#include <math.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/**
+ * Calculate the L2 norm of the vector at the given t-value
```

Comment:
maybe not the best place to ask, but I don't see why the L2 norm of a point on the Poly2D is useful?

Reply:
You need it for the arc length integral. https://en.wikipedia.org/wiki/Arc_length